ci_framework/plugins/action/oc.py

```diff
@@ -0,0 +1,113 @@
+# Copyright Red Hat, Inc.
+# Apache License Version 2.0 (see LICENSE)
+
+import os
+from pathlib import Path
+from ansible.errors import AnsibleError
+from ansible.module_utils.basic import get_bin_path
+from ansible.module_utils.six import ensure_text
+from ansible.plugins.action import ActionBase
+
+DOCUMENTATION = r'''
+---
+action: oc
+
+short_description: Execute OpensShift Client commands - oc -.
+
+description:
+- This module searches the oc binary, load the kubeconfig and execute the oc provided commands.
+'''
+
+EXAMPLES = r'''
+- name: Display the current authenticated user
+    oc_cmd:
+    command: whoami -c
+- name: Display the current authenticated user and get cluster information
+      oc:
+        cmd: |
+          whoami -c
+          cluster-info
+'''
+
+RETURN = r'''
+success:
+    description: Status of the module
+    type: bool
+    returned: always
+    sample: True
+stdout:
+    description: List with the command execution result/s
+    type: list
+    returned: always
+'''
+
+
+class ActionModule(ActionBase):
+
+    def run(self, tmp=None, task_vars=None):
+        super().run(tmp, task_vars)
+        module_args = self._task.args.copy()
+
+        if 'cmd' not in module_args or not module_args['cmd']:
+            raise AnsibleError('"cmd" parameter is mandatory')
+
+        # Get remote home, local or remote depending of the hosts
+        user_home = Path(task_vars['ansible_user_dir']).resolve()
+
+        # Get cluster auth config path or take by default crc one
+        kubeconfig_path = task_vars.get(
+            'cifmw_kubeconfig', False) or user_home / ".crc/machines/crc/kubeconfig"
+        os.environ["KUBECONFIG"] = str(kubeconfig_path)
```

Comment:
This may work for localhost, but not for remotes. Please check `_compute_environment_string` and `_execute_module` https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/action/__init__.py#L1000